estimagic/optimization/optimize.py

```diff
@@ -418,3 +421,35 @@ def _process_results(res, params, internal_params, constraints, origin):
 
     res_dict["internal_x"] = x.tolist()
     return res_dict, params
+
+
+def _logging(params, fitness_eval, counter, log_interval=100, path_to_sql="logging.db"):
+    """Log parameter values and fitness.
+
+    Saves optimization results every `log_interval` iteration to `path_to_sql`.
+
+    Arg:
+        params (DataFrame): See :ref:`params`
+        fitness_eval (float): current fitness evaluation
+        counter (int): number of current iteration
+        log_interval (int): number of iterations until first logging shall
```

Comment:
The more important feature here seems to be the spacing between triggering a log entry rather than when it starts.

Default should be 1 not 100, IMHO. Unless errors / invalid values mean things are logged post mortem, but that does not seem to be the case? [one usually wants to know from the logs where things went wrong, does not help if it is in between the intervals]

Reply:
> 
> 
> General: Please leave the connection open all the time. I guess that would require splitting this up into "start_log", "do_log", "finish_log".

well, I guess this needs to be done at a different point in the code? Because, I do not see how the functoin is supposed to know when the optimization is finished...

Reply:
Can I ask why this is important?l is this a speed issue?

Reply:
Yes. Most of the overhead comes from there.

Janos is completely right about the context manager, of course. In fact, I bet you can write `with sqlite.connect(path_to_db) as conn:`, at least that is how it works with the db connectors I have worked with. The logging function would then get the connection. Less happening inside of it, good.

Reply:
yeah there is something like that in the documentation, but still not sure, what exactly you want from me :), maybe a little bit more ellaboration for noobs like me,

Reply:
Is the below clear?